typespec-tests/src/main/java/com/cadl/optional/OptionalAsyncClient.java

```diff
@@ -55,6 +55,8 @@ public final class OptionalAsyncClient {
      * <table border="1">
      * <caption>Header Parameters</caption>
      * <tr><th>Name</th><th>Type</th><th>Required</th><th>Description</th></tr>
+     * <tr><td>Content-Type</td><td>String</td><td>No</td><td>The content type. Allowed values:
+     * "application/json".</td></tr>
```

Comment:
This appears to be actually a bug fix.

The request body is optional, hence we actually ask user to set the `content-type` header, even this is just 1 content-type.
```java
                if (MethodUtil.getContentTypeCount(operation.getRequests()) == 1 && !isBodyParamRequired) {
                    Parameter contentTypeParameter = MethodUtil.createContentTypeParameter(request, operation);
                    allParameters.add(Mappers.getProxyParameterMapper().map(contentTypeParameter));
                }
```